activerecord/lib/active_record/relation/finder_methods.rb

```diff
@@ -533,6 +533,12 @@ def find_nth_with_limit(index, limit)
           relation = ordered_relation
 
           if limit_value.nil? || index < limit_value
+
+            if limit_value.present?
+              limit = [limit_value - index, limit].min
+              limit = [limit, 0].max
```

Comment:
If `limit <= 0`, we don't need a query to database.

```diff
diff --git a/activerecord/lib/active_record/relation/finder_methods.rb b/activerecord/lib/active_record/relation/finder_methods.rb
index e61cacf6a7..50d0f14b98 100644
--- a/activerecord/lib/active_record/relation/finder_methods.rb
+++ b/activerecord/lib/active_record/relation/finder_methods.rb
@@ -532,7 +532,11 @@ def find_nth_with_limit(index, limit)
         else
           relation = ordered_relation
 
-          if limit_value.nil? || index < limit_value
+          if limit_value
+            limit = [limit_value - index, limit].min
+          end
+
+          if limit > 0
             relation = relation.offset(offset_index + index) unless index.zero?
             relation.limit(limit).to_a
           else
```

Reply:
Great point! I have updated the PR.